BehavioralPatterns/BehavioralPatterns/SchoolList.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace BehavioralPatterns
+{
+    class SchoolList : IEnumerable<int>
+    {
+        private List<School> schools = new List<School>();
```

Comment:
You should use interfaces of bse clases for references. IList should work fot this case.